src/pandas_profiling/config.py

```diff
@@ -185,6 +189,22 @@ class Html(BaseModel):
 
     full_width: bool = False
 
+class JsonNonFiniteEncoding(Enum):
+    # Use the default python behaviour, which violates the official JSON standard, basically allow_nan = False
+    __default = 0
+    # Encode non-finite numbers as null values, allow_nan = True
+    __num_null = 1
+    # Encode non-finite floats as null values, allow_nan = True
+    __float_null = 2
+
+    def fetch_python(self):
+        return self.__default
+
+    def fetch_null_values(self):
+        return self.__num_null
+
+    def fetch_float_values(self):
+        return self.__float_null
```

Comment:
I believe Lukas' version is a bit more clear in this part. Can you help me understand why we would want private fields instead of Lukas' version?